R/utils.R

```diff
@@ -31,3 +34,37 @@ get_ppm_snapshot_by_date <- function(date) {
 parse_ppm_url <- function(snapshot) {
   file.path(pkgcache::ppm_repo_url(), snapshot)
 }
+
+#' Resolve the dependencies of package based on the release date + 1
+#'
+#' @keywords internal
+#' @importFrom pkgcache ppm_repo_url
+#' @importFrom pkgdepends new_pkg_deps parse_pkg_ref
+resolve_ppm_snapshot <- function(pkg_ref_str, operator, pkg_version) {
+
+  i_ref <- pkgdepends::parse_pkg_ref(pkg_ref_str)
+
+  i_ref_minver <- get_ref_min_incl_cran(i_ref, operator, pkg_version)
+
+  i_release_date <- get_release_date(i_ref_minver)
+
+  if (all(is.na(i_release_date))) {
+    ppm_repo <- file.path(pkgcache::ppm_repo_url(), "latest")
+  } else {
+    ppm_repo <- parse_ppm_url(get_ppm_snapshot_by_date(i_release_date))
+  }
+
+  i_pkg_deps <- pkgdepends::new_pkg_deps(
+    ifelse(
+      inherits(i_ref_minver, "remote_ref_github"),
+      i_ref_minver$ref,
+      i_ref$ref
+    ),
+    config = list(dependencies = "hard", cran_mirror = ppm_repo)
+  )
```

Comment:
We need to use an empty libpath without already installed packages

```suggestion
  config <- append_config(default_config(), list(dependencies = "hard", cran_mirror = ppm_repo))

  i_pkg_deps <- pkgdepends::new_pkg_deps(
    ifelse(
      inherits(i_ref_minver, "remote_ref_github"),
      i_ref_minver$ref,
      i_ref$ref
    ),
    config = config
  )
```